### PR TITLE
chore: Convert example `2-with-_app.js` to TypeScript

### PR DESCRIPTION
**Pull request by 0xRAG**
### Summary

This PR converts `2-with-_app.js` to use TypeScript (https://github.com/FormidableLabs/next-urql/issues/18). It follows the same strategy used in [converting the first example](https://github.com/FormidableLabs/next-urql/pull/19).

### CODE REVIEW

File: examples/2-with-_app.js/pages/_app.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { withUrqlClient } from 'next-urql';
+
+interface IApp {
+  Component: React.FC;
+  pageProps: Record<string, any>;
+}
+
+const App: React.FC<IApp> = ({ Component, pageProps }) => (
```
Comment (parkerziegler) [inline examples/2-with-_app.js/pages/_app.tsx]: If I'm not mistaken, I believe `next` has a type specifically for props handed to `_app.js` components: https://github.com/zeit/next.js/blob/58b2d9e208c56cee0dab2eff287d2a19cb0e6de1/packages/next/next-server/lib/utils.ts#L129-L135. It's not super precise, but let's use there type signature here:

```suggestion
const App: React.FC<AppPropsType> = ({ Component, pageProps }) => (
```

**Review by parkerziegler — APPROVED** [review]
You're crushing it @ryan-gilb. Just one small recommendation about bringing in the `AppPropsType` from `next`. Thanks as well for opening the issue tracking upgrading `next` types. If you feel like tackling that one I'm happy to review it, otherwise I can get to it sometime in the next day or so.

File: examples/2-with-_app.js/pages/_app.tsx (hunk 1 above)
Comment (0xRAG) [reply examples/2-with-_app.js/pages/_app.tsx]: Oh cool, I didn't see that! Updating now

**Comment by 0xRAG** [comment]
Thanks @parkerziegler ❤️ Good catch on `AppPropsType`, I updated the example to use it. This should be ready to merge!

I'm happy to work on the issue tracking upgrading `next` types 😄

**Comment by parkerziegler** [comment]
:heck: yes, awesome work @ryan-gilb and thanks for taking on getting the types updated!

**Pull request merged**